tests/phpunit/integration/Modules/Reader_Revenue_Manager/Tag_GuardTest.php

```diff
@@ -29,34 +29,167 @@ class Tag_GuardTest extends TestCase {
 	 */
 	private $settings;
 
-	/**
-	 * Tag_Guard object.
-	 *
-	 * @var Tag_Guard
-	 */
-	private $guard;
-
 	public function set_up() {
 		parent::set_up();
 
-		update_option(
-			Settings::OPTION,
+		$this->settings = new Settings( new Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ) );
+
+		$this->settings->register();
+		$this->settings->merge(
 			array(
 				'publicationID' => '12345',
 			)
 		);
-
-		$this->settings = new Settings( new Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) ) );
-		$this->guard    = new Tag_Guard( $this->settings );
 	}
 
 	public function test_can_activate() {
-		$this->assertTrue( $this->guard->can_activate() );
+		$guard = new Tag_Guard( $this->settings, '' );
+
+		$this->assertTrue( $guard->can_activate() );
 	}
 
-	public function test_cant_activate_when_usesnippet_is_falsy() {
+	public function test_can_not_activate_when_publication_id_is_unset() {
 		$this->settings->merge( array( 'publicationID' => '' ) );
 
-		$this->assertFalse( $this->guard->can_activate(), 'Should return FALSE when publicationID is not set.' );
+		$guard = new Tag_Guard( $this->settings, '' );
+
+		$this->assertFalse(
+			$guard->can_activate(),
+			'should return FALSE when publicationID is not set.'
+		);
+	}
+
+	public function data_configurations__singular() {
+		return array(
+			'no publication id'         => array(
+				array(
+					'publicationID' => '',
+				),
+				'',
+				false,
+			),
+			'with post-product-id none' => array(
+				array(
+					'publicationID' => '12345',
+				),
+				'none',
+				false,
+			),
+			'with post-product-id'      => array(
+				array(
+					'publicationID' => '12345',
+				),
+				'12345',
+				true,
+			),
+			'with empty post-product-id and snippet mode of per post' => array(
+				array(
+					'publicationID' => '12345',
+					'snippetMode'   => 'per_post',
+				),
+				'',
+				false,
+			),
+			'with empty post-product-id and snippet mode of post types' => array(
+				array(
+					'publicationID' => '12345',
+					'snippetMode'   => 'post_types',
+				),
```

Comment:
This confused me for a moment, let's add a comment to clarify it:

```suggestion
					// The `post` type is in the `postTypes` setting by default.
				),
```

Reply:
Added, thank you!